synapse/http/server.py

```diff
@@ -79,10 +81,12 @@ def return_json_error(f: failure.Failure, request: SynapseRequest) -> None:
     """Sends a JSON error response to clients."""
 
     if f.check(SynapseError):
```

Comment:
Should we change this to `isinstance(f.value, SynapseError)` perhaps?

Reply:
I don't think so. This is the proper way to check against twisted failures?

Reply:
True, probably easier (and more likely) for us to remove a `# type: ignore` in the future than switch away from an `isinstance`.

Reply:
I think my thought is that if we want to make that change we should do is as a discrete step, not bury it in this PR!